sim/pkg/plugin/plugin.go

```diff
@@ -0,0 +1,46 @@
+package plugin
+
+import (
+	"github.com/josephburnett/sk-plugin/pkg/skplug"
+	"github.com/josephburnett/sk-plugin/pkg/skplug/dispatcher"
+	"github.com/josephburnett/sk-plugin/pkg/skplug/proto"
+	"strconv"
+	"sync/atomic"
+)
+
+type PluginPartition interface {
+	Event(time int64, typ proto.EventType, object skplug.Object) error
+	Stat(stat []*proto.Stat) error
+	HorizontalRecommendation(time int64) (rec int32, err error)
+	VerticalRecommendation(time int64) (rec []*proto.RecommendedPodResources, err error)
+}
+
+type pluginPartition struct {
+	partition  string
+	dispatcher dispatcher.Dispatcher
+}
+
+var partitionSequence int32 = 0
+
+func NewPluginPartition() PluginPartition {
+	return &pluginPartition{
+		partition:  strconv.Itoa(int(atomic.AddInt32(&partitionSequence, 1))),
+		dispatcher: dispatcher.GetInstance(),
```

Comment:
I would prefer to take a dispatcher as input to `NewPluginPartition`, which comes all the way from a single instance created by main.  Any objections?